apps/charterafrica/src/lib/openAfrica.js

```diff
@@ -1,84 +1,131 @@
+import allCountries from "@/charterafrica/lib/data/json/countries";
 import fetchJson from "@/charterafrica/utils/fetchJson";
 
 const BASE_DOCUMENTS_URL = "https://openafrica.net/api/3/action/";
 
-async function formatDatasets(data) {
-  const {
-    result: {
-      count,
-      facets: { tags: allTags, groups: allCountries },
-      results,
-    },
-  } = data || {};
+const PAGE_SIZE = 10;
 
-  const formattedDatasets = results?.map((dataset) => {
+async function packageSearch(params) {
+  try {
+    const response = await fetchJson.get(
+      `${BASE_DOCUMENTS_URL}package_search`,
+      {
+        params,
+      }
+    );
+    return response;
+  } catch (error) {
+    return error;
+  }
+}
+
+function formatResources(resources, author) {
+  return resources?.map((resource) => {
+    const {
+      created,
+      id,
+      last_modified: updated,
+      url,
+      format,
+      name,
+      description,
+    } = resource;
+    return {
+      author,
+      created,
+      description,
+      format,
+      id,
+      name,
+      updated,
+      url,
+    };
+  });
+}
+
+function formatDatasets(datasets) {
+  return datasets?.map((dataset) => {
     const {
       author,
+      id,
       metadata_created: created,
       metadata_modified: updated,
       name,
       notes,
       resources,
       title,
       type,
+      url,
     } = dataset;
 
-    const formattedResources = resources?.map((resource) => {
-      const { url, format, name: resourceName, description } = resource;
-      return {
-        description,
-        format,
-        name: resourceName,
-        url,
-      };
-    });
+    const formattedResources = formatResources(resources, author);
 
-    const allDocumentFormats = [
-      ...new Set(formattedResources?.map((resource) => resource.format)),
-    ];
+    const formatsSet = new Set(resources?.map((r) => r.format));
+    const formats = Array.from(formatsSet);
```

Comment:
```suggestion
    const formats = Array.from(formatsSet).sort();
```
Small thing but I think it will better if we consistently show formats in one order e.g. CSV, PDF or PDF, CSV every time there is more than one format. Mixing them up doesn't really look... neat.